<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Compare the performance of peco with other methods using Leng et al. 2015 data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">peco-paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jhsiao999/peco-paper">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Compare the performance of peco with other methods using Leng et al. 2015 data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>peco-paper/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190814code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190814)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190814code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190814)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjhsiao999pecopapertreecfbce75043ab562c67f335b49ffa9716f08c7c6btargetblankcfbce75a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jhsiao999/peco-paper/tree/cfbce75043ab562c67f335b49ffa9716f08c7c6b" target="_blank">cfbce75</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjhsiao999pecopapertreecfbce75043ab562c67f335b49ffa9716f08c7c6btargetblankcfbce75a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/npreg_trendfilter_quantile.Rmd
    Untracked:  data/HumanLengESC.rds
    Untracked:  data/data_training_test/
    Untracked:  data/eset-filtered.rds
    Untracked:  data/fit.quant.rds
    Untracked:  data/fit.trend.perm.lowmiss.rds
    Untracked:  data/fit_diff_cyclone.rds
    Untracked:  data/fit_diff_oscope.rds
    Untracked:  data/fit_diff_peco.rds
    Untracked:  data/fit_diff_recat.rds
    Untracked:  data/fit_diff_seurat.rds
    Untracked:  data/intensity.rds
    Untracked:  data/leng2015_data.rds
    Untracked:  data/leng_fucci_oscope_29genes.rda
    Untracked:  data/leng_fucci_recat.rda
    Untracked:  data/leng_geneinfo.txt
    Untracked:  data/log2cpm.quant.rds
    Untracked:  data/macosko-2015.rds
    Untracked:  data/nmeth.3549-S2.xlsx
    Untracked:  data/ourdata_cyclone_NA18511.rds
    Untracked:  data/ourdata_cyclone_NA18855.rds
    Untracked:  data/ourdata_cyclone_NA18870.rds
    Untracked:  data/ourdata_cyclone_NA19098.rds
    Untracked:  data/ourdata_cyclone_NA19101.rds
    Untracked:  data/ourdata_cyclone_NA19160.rds
    Untracked:  data/ourdata_oscope_366genes.rda
    Untracked:  data/ourdata_peco_NA18511_top005genes.rds
    Untracked:  data/ourdata_peco_NA18855_top005genes.rds
    Untracked:  data/ourdata_peco_NA18870_top005genes.rds
    Untracked:  data/ourdata_peco_NA19098_top005genes.rds
    Untracked:  data/ourdata_peco_NA19101_top005genes.rds
    Untracked:  data/ourdata_peco_NA19160_top005genes.rds
    Untracked:  data/ourdata_phase_cyclone.rds
    Untracked:  data/ourdata_phase_seurat.rds
    Untracked:  data/ourdata_recat.rda
    Untracked:  data/sce-filtered.rds

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   code/run_seurat.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jhsiao999/peco-paper/blob/cfbce75043ab562c67f335b49ffa9716f08c7c6b/analysis/eval_on_leng2015_data.Rmd" target="_blank">cfbce75</a>
</td>
<td>
jhsiao999
</td>
<td>
2020-01-26
</td>
<td>
Leng et al. 2015 data results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In Leng et al. 2015, FUCCI reporters were used to sort single-cell samples (Human Embryonic Stem Cells) into G1, S and G2M phases. The single-cell samples in each phase were then captured separated in 3 Fluidigm C1 plates.</p>
<p>We used this datat to assess the ability of peco to predict cell cycle phase against a “ground truth”. Specifically, we compared the performance of peco with Oscope and reCAT in predicting continuous cell cycle phase. We also compared the reuslts of Seurat and Oscope with the Leng et al. gating-based classification and our PAM-based discrete classification.</p>
</div>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<p>Load packages</p>
<pre class="r"><code>library(SingleCellExperiment)
library(peco)
library(yarrr)
library(tidyverse)
library(SingleCellExperiment)
library(peco)</code></pre>
<p>Getting data</p>
<pre class="r"><code># ------ getting data
sce &lt;- readRDS(&quot;data/leng2015_data.rds&quot;)
pdata &lt;- data.frame(colData(sce))
fdata &lt;- data.frame(rowData(sce))
counts &lt;- assay(sce)

sce_fucci &lt;- sce[,pdata$cell_state != &quot;H1&quot;]

# compute quantile-normalized gene expression levels
sce_fucci &lt;- data_transform_quantile(sce_fucci)</code></pre>
<pre><code>computing on 2 cores</code></pre>
<pre class="r"><code>counts_fucci &lt;- assay(sce_fucci, &quot;counts&quot;)
cpm_quantNormed_fucci &lt;- assay(sce_fucci, &quot;cpm_quantNormed&quot;)

pdata_fucci &lt;- data.frame(colData(sce_fucci))
pdata_fucci$cell_state &lt;- droplevels(pdata_fucci$cell_state)
pdata_fucci$cell_state &lt;- factor(pdata_fucci$cell_state,
                                      levels = c(&quot;G1&quot;, &quot;S&quot;, &quot;G2&quot;),
                                      labels = c(&quot;G1&quot;, &quot;S&quot;, &quot;G2M&quot;))</code></pre>
</div>
<div id="peco-prediction" class="section level2">
<h2>peco prediction</h2>
<pre class="r"><code># load training data
library(peco)
data(&quot;sce_top101genes&quot;)
data(&quot;training_human&quot;)


# Use top 4 genes for HIST1H4E was deprecated in genome build hg38
gene_symbols &lt;- c(&quot;CDK1&quot;, &quot;UBE2C&quot;, &quot;TOP2A&quot;, &quot;HIST1H4C&quot;)
gene_ensg &lt;- c(&quot;ENSG00000170312&quot;,&quot;ENSG00000175063&quot;,
           &quot;ENSG00000131747&quot;, &quot;ENSG00000197061&quot;)
# use top 101 genes to predict
training_fun = training_human$cellcycle_function[gene_ensg]
training_sigma = training_human$sigma[gene_ensg,]


# use get_trend_estimateds=TRUE to estimate cyclic expression levels of 
# the four genes in Leng et al. 2015 data
out_peco &lt;- cycle_npreg_outsample(
  Y_test=sce_fucci[rownames(sce_fucci) %in% gene_symbols],
  funs_est=training_fun,
  sigma_est=training_sigma,
  method.trend=&quot;trendfilter&quot;,
  ncores=4,
  get_trend_estimates=TRUE)</code></pre>
<pre><code>computing on 4 cores</code></pre>
<pre class="r"><code>subdata_plot &lt;- do.call(rbind, lapply(1:4, function(g) {
  gindex &lt;- which(rownames(out_peco$Y_reordered) == gene_symbols[g])
  gexp &lt;- out_peco$Y_reordered[gindex,]
  data.frame(gexp=gexp, gene=gene_symbols[g], cell_state=pdata_fucci$cell_state)
}))
ggplot(subdata_plot, aes(x=cell_state, y = gexp, fill=cell_state)) +
  geom_boxplot() + facet_wrap(~gene, ncol=4) +
  ylab(&quot;Quantile-normalized \n gene expression levels&quot;) + xlab(&quot;&quot;) +
  scale_fill_manual(values=as.character(yarrr::piratepal(&quot;espresso&quot;)[3:1]),
                    name=&quot;Phase&quot;) + theme_bw()</code></pre>
<p><img src="figure/eval_on_leng2015_data.Rmd/peco-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,4), mar=c(2,2,2,1), oma = c(0,0,2,0))

all.equal(colnames(out_peco$Y_reordered), rownames(pdata_fucci))</code></pre>
<pre><code>[1] &quot;246 string mismatches&quot;</code></pre>
<pre class="r"><code>cell_state_col &lt;- data.frame(cell_state = pdata_fucci$cell_state[match(colnames(out_peco$Y_reordered),
                                                                            rownames(pdata_fucci))],
                             sample_id = pdata_fucci$sample_id[match(colnames(out_peco$Y_reordered),
                                                                            rownames(pdata_fucci))])
cell_state_col$cols &lt;- as.character(yarrr::piratepal(&quot;espresso&quot;)[3:1])[cell_state_col$cell_state]

for (g in 1:4) {
  if (g&gt;1) { ylab &lt;- &quot;&quot;} else { ylab &lt;- &quot;Quantile-normalized expression value&quot;}
  #    if (i==1) { xlab &lt;- &quot;Predicted phase&quot;} else { xlab &lt;- &quot;Inferred phase&quot;}
  gindex &lt;- which(rownames(out_peco$Y_reordered) == gene_symbols[g])

  plot(x= out_peco$cell_times_reordered,
       y = out_peco$Y_reordered[gindex,], pch = 16, cex=.5, ylab=ylab,
       xlab=&quot;Predicted phase&quot;, col = cell_state_col$cols,
       main = gene_symbols[g], axes=F, ylim=c(-3,3))
  abline(h=0, lwd=.5)
  lines(x = seq(0, 2*pi, length.out=200),
        y = out_peco$funs_reordered[[g]](seq(0, 2*pi, length.out=200)),
        col=wesanderson::wes_palette(&quot;FantasticFox1&quot;)[1], lty=1, lwd=2)
  axis(2); axis(1,at=c(0,pi/2, pi, 3*pi/2, 2*pi),
                labels=c(0,expression(pi/2), expression(pi), expression(3*pi/2),
                         expression(2*pi)))
}
title(&quot;peco prediction results&quot;, outer = TRUE, line = .5)</code></pre>
<p><img src="figure/eval_on_leng2015_data.Rmd/peco-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="oscope-prediction" class="section level2">
<h2>Oscope prediction</h2>
<p>We applied Oscope to estimate cell cycle ordering across the 888 single-cell samples. The analysis used the 29 genes that were identified as oscillating cell cycle genes in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4589503/">Leng et al. 2015</a> (<a href="doi:10.1038/nmeth.3549" class="uri">doi:10.1038/nmeth.3549</a>). The data file was downloaed from nmeth.3549-S2.xlsx.</p>
<pre class="r"><code># Oscope
library(openxlsx)
oscope29genes &lt;- unlist(read.xlsx(&quot;data/nmeth.3549-S2.xlsx&quot;, colNames = F))

# run Oscope on FUCCI cells ------------------------------
library(Oscope)
Sizes &lt;- MedianNorm(counts_fucci)
DataNorm &lt;- GetNormalizedMat(counts_fucci, Sizes)
DataNorm_sub &lt;- DataNorm[which(rownames(DataNorm) %in% oscope29genes),]
DataInput &lt;- NormForSine(DataNorm_sub)
SineRes &lt;- OscopeSine(DataInput, parallel=F)
KMResUse &lt;- list(cluster1=oscope29genes)
ENIRes &lt;- OscopeENI(KMRes = KMResUse, Data = DataInput,
                    NCThre = 100, parallel=T)

save(DataInput,
     KMResUse, ENIRes, fits_leng_oscope, pdata_fucci,
     file = &quot;data/leng_fucci_oscope_29genes.rda&quot;)</code></pre>
<pre class="r"><code># load pre-computed results,
load(&quot;data/leng_fucci_oscope_29genes.rda&quot;)

# lot predicted results
gene_symbols &lt;- c(&quot;CDK1&quot;, &quot;UBE2C&quot;, &quot;TOP2A&quot;, &quot;HIST1H4C&quot;)
gene_ensg &lt;- c(&quot;ENSG00000170312&quot;,&quot;ENSG00000175063&quot;,
           &quot;ENSG00000131747&quot;, &quot;ENSG00000197061&quot;)

phase_pred &lt;- seq(0, 2*pi, length.out= length(ENIRes$cluster1))
names(phase_pred) &lt;- colnames(DataInput)[ENIRes[[&quot;cluster1&quot;]]]

fits_leng_oscope &lt;- lapply(1:4, function(g) {
  gindex &lt;- which(rownames(cpm_quantNormed_fucci) == gene_symbols[g])
  fit_g &lt;- data.frame(
    gexp=cpm_quantNormed_fucci[gindex, match(names(phase_pred), colnames(cpm_quantNormed_fucci))],
    phase=shift_origin((-phase_pred+2*pi), origin = pi/2))
  fit_g &lt;- fit_g[order(fit_g$phase),]

  fit_trend &lt;- fit_trendfilter_generic(fit_g$gexp)
  fit_g$trend.yy &lt;- fit_trend$trend.yy
  fun_g &lt;- approxfun(x=as.numeric(fit_g$phase),
                     y=as.numeric(fit_g$trend.yy), rule=2)
  fit_out &lt;- list(fit_g=fit_g,
                  pve = fit_trend$pve,
                  fun_g = fun_g)
  return(fit_out)
})</code></pre>
<pre><code>Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... 
Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... 
Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... 
Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... </code></pre>
<pre class="r"><code>names(fits_leng_oscope) &lt;- gene_symbols


all.equal(rownames(fits_leng_oscope[[1]]$fit_g), rownames(pdata_fucci))</code></pre>
<pre><code>[1] &quot;247 string mismatches&quot;</code></pre>
<pre class="r"><code>cell_state_oscope_col &lt;- 
    data.frame(cell_state = pdata_fucci$cell_state[match(rownames(fits_leng_oscope[[1]]$fit_g),
                                               rownames(pdata_fucci))],
              sample_id = rownames(fits_leng_oscope[[1]]$fit_g) )
cell_state_oscope_col$cols &lt;-
    as.character(yarrr::piratepal(&quot;espresso&quot;)[3:1])[cell_state_oscope_col$cell_state]



par(mfrow=c(1,4), mar=c(2,2,2,1), oma = c(0,0,2,1))
for (g in 1:4) {
  if (g&gt;1) { ylab &lt;- &quot;&quot;} else { ylab &lt;- &quot;Normalized log2CPM&quot;}
  #    if (i==1) { xlab &lt;- &quot;Predicted phase&quot;} else { xlab &lt;- &quot;Inferred phase&quot;}
  res_g &lt;- fits_leng_oscope[[g]]
  plot(x= res_g$fit_g$phase,
       y = res_g$fit_g$gexp, pch = 16, cex=.5, ylab=ylab,
       xlab=&quot;Predicted phase&quot;, col=cell_state_oscope_col$cols,
       main = names(fits_leng_oscope)[g], axes=F, ylim=c(-3,3))
  abline(h=0, lwd=.5)
  lines(x = seq(0, 2*pi, length.out=200),
        y = res_g$fun_g(seq(0, 2*pi, length.out=200)),
        col = wesanderson::wes_palette(&quot;Darjeeling2&quot;)[2], lty=1, lwd=2)
  axis(2); axis(1,at=c(0,pi/2, pi, 3*pi/2, 2*pi),
                labels=c(0,expression(pi/2), expression(pi), expression(3*pi/2),
                         expression(2*pi)))
}
title(&quot;Oscope prediction results&quot;, outer = TRUE, line = .5)</code></pre>
<p><img src="figure/eval_on_leng2015_data.Rmd/plot-oscope-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="recat-prediction" class="section level2">
<h2>reCAT prediction</h2>
<p>We applied reCAT to estimate cell cycle ordering across the 888 single-cell samples. The analysis used all the 11,040 genes.</p>
<p>To run recAT, please clone the reCAT GitHUb repository and then cd to the directory.</p>
<p><code>git clone https://github.com/tinglab/reCAT</code></p>
<p><code>cd &quot;reCAT/R&quot;</code></p>
<pre class="r"><code># recat
input &lt;- t(cpm_quantNormed_fucci)
input &lt;- input[sample(1:nrow(input)),]
source(&quot;get_test_exp.R&quot;)
test_exp &lt;- get_test_exp(t(input))
source(&quot;get_ordIndex.R&quot;)
res_ord &lt;- get_ordIndex(test_exp, 10)
ordIndex &lt;- res_ord$ordIndex

save(test_exp, res_ord, ordIndex, &quot;data/leng_fucci_recat.rda&quot;)</code></pre>
<pre class="r"><code># load pre-computed results
load(&quot;data/leng_fucci_recat.rda&quot;)

sample_ordered &lt;- rownames(test_exp)[ordIndex]
phase_pred &lt;- seq(0, 2*pi, length.out= length(ordIndex))
names(phase_pred) &lt;- sample_ordered

fits_recat &lt;- lapply(1:4, function(g) {
  gindex &lt;- which(rownames(cpm_quantNormed_fucci) == gene_symbols[g])
  fit_g &lt;- data.frame(
    gexp=cpm_quantNormed_fucci[gindex, match(sample_ordered, colnames(cpm_quantNormed_fucci))],
    phase=(-phase_pred)+2*pi )
  fit_g &lt;- fit_g[order(fit_g$phase),]

  fit_trend &lt;- fit_trendfilter_generic(fit_g$gexp, polyorder = 2)
  fit_g$trend.yy &lt;- fit_trend$trend.yy
  #    fit_g$pve &lt;- fit$pve
  fun_g &lt;- approxfun(x=as.numeric(fit_g$phase),
                     y=as.numeric(fit_g$trend.yy), rule=2)
  fit_out &lt;- list(fit_g=fit_g,
                  #                  pve = fit$pve,
                  fun_g = fun_g)
  return(fit_out)
})</code></pre>
<pre><code>Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... 
Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... 
Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... 
Fold 1 ... Fold 2 ... Fold 3 ... Fold 4 ... Fold 5 ... </code></pre>
<pre class="r"><code>names(fits_recat) &lt;- gene_symbols

all.equal(rownames(fits_recat[[1]]$fit_g), rownames(pdata_fucci))</code></pre>
<pre><code>[1] &quot;246 string mismatches&quot;</code></pre>
<pre class="r"><code>cell_state_recat_col &lt;- 
    data.frame(cell_state = pdata_fucci$cell_state[match(rownames(fits_recat[[1]]$fit_g),
                                                       rownames(pdata_fucci))],
               sample_id = rownames(fits_recat[[1]]$fit_g) )
cell_state_recat_col$cols &lt;- 
    as.character(yarrr::piratepal(&quot;espresso&quot;)[3:1])[cell_state_recat_col$cell_state]

par(mfrow=c(1,4), mar=c(2,2,2,1), oma = c(0,0,2,0) )
for (g in 1:4) {
  if (g&gt;1) { ylab &lt;- &quot;&quot;} else { ylab &lt;- &quot;Quantile-normalized expression value&quot;}
  res_g &lt;- fits_recat[[g]]
  plot(x= res_g$fit_g$phase,
       y = res_g$fit_g$gexp, pch = 16, cex=.5, ylab=ylab,
       xlab=&quot;Predicted phase&quot;, col = cell_state_recat_col$cols,
       main = names(fits_recat)[g], axes=F, ylim=c(-3,3))
  abline(h=0, lwd=.5)
  lines(x = seq(0, 2*pi, length.out=200),
        y = res_g$fun_g(seq(0, 2*pi, length.out=200)),
        col=wesanderson::wes_palette(&quot;FantasticFox1&quot;)[5], lty=1, lwd=2)
  axis(2); axis(1,at=c(0,pi/2, pi, 3*pi/2, 2*pi),
                labels=c(0,expression(pi/2), expression(pi), expression(3*pi/2),
                         expression(2*pi)))
}
title(&quot;reCAT prediction results&quot;, outer = TRUE, line = .5)</code></pre>
<p><img src="figure/eval_on_leng2015_data.Rmd/plot-recar-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="seurat" class="section level2">
<h2>Seurat</h2>
<pre class="r"><code>library(Seurat)</code></pre>
<pre><code>
Attaching package: &#39;Seurat&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:SummarizedExperiment&#39;:

    Assays</code></pre>
<pre class="r"><code>cc.genes &lt;- readLines(con = &quot;data//regev_lab_cell_cycle_genes.txt&quot;)

obj &lt;- CreateSeuratObject(counts = counts_fucci)
obj &lt;- NormalizeData(obj)
obj &lt;- FindVariableFeatures(obj, selection.method = &quot;vst&quot;)
obj &lt;- ScaleData(obj, features = rownames(obj))</code></pre>
<pre><code>Centering and scaling data matrix</code></pre>
<pre class="r"><code>obj &lt;- CellCycleScoring(obj, s.features = cc.genes[1:43], 
                        g2m.features = cc.genes[44:97], set.ident = TRUE)
out_seurat &lt;- obj[[]]

all.equal(rownames(out_seurat), colnames(counts_fucci))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>out_seurat &lt;- out_seurat[match(colnames(counts_fucci), rownames(out_seurat)),]

out_seurat$Phase &lt;- factor(out_seurat$Phase,
                           levels = c(&quot;G1&quot;, &quot;S&quot;, &quot;G2M&quot;))
out_seurat$gates &lt;- pdata_fucci$cell_state[match(rownames(out_seurat),
                                                       rownames(pdata_fucci))]
table(out_seurat$gates, out_seurat$Phase)</code></pre>
<pre><code>     
      G1  S G2M
  G1  10 28  53
  S   20 60   0
  G2M 11  5  60</code></pre>
</div>
<div id="cyclone" class="section level2">
<h2>Cyclone</h2>
<pre class="r"><code>library(scran)
hs.pairs &lt;- readRDS(system.file(&quot;exdata&quot;, &quot;human_cycle_markers.rds&quot;, package=&quot;scran&quot;))

## get ENSG ID
# library(biomaRt)
# mart &lt;- useMart(biomart = &quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)
# geneinfo &lt;- getBM(attributes = c(&quot;hgnc_symbol&quot;, &quot;ensembl_gene_id&quot;),
#                filters = &quot;hgnc_symbol&quot;,
#                values = rownames(cpm_quantNormed_fucci), bmHeader = T, mart = mart)
#write.table(geneinfo, file = &quot;data/leng_geneinfo.txt&quot;))

geneinfo &lt;- read.table(&quot;data/leng_geneinfo.txt&quot;)


indata &lt;- which(geneinfo$ensg %in% unlist(hs.pairs))
geneinfo_indata &lt;- geneinfo[indata,]

Y_cyclone &lt;- cpm_quantNormed_fucci[rownames(cpm_quantNormed_fucci) %in% as.character(geneinfo_indata$symbol),]
rownames(Y_cyclone) &lt;- geneinfo_indata$ensg[match(rownames(Y_cyclone),
                                                  geneinfo_indata$symbol)]

out_cyclone &lt;- cyclone(Y_cyclone, pairs = hs.pairs,
                       gene.names=rownames(Y_cyclone),
                       iter=1000, min.iter=100, min.pairs=50,
                       BPPARAM=SerialParam(), verbose=T, subset.row=NULL)</code></pre>
<pre><code>Number of G1 pairs: 21666</code></pre>
<pre><code>Number of S pairs: 26446</code></pre>
<pre><code>Number of G2M pairs: 18738</code></pre>
<pre class="r"><code>names(out_cyclone$phases) &lt;- colnames(Y_cyclone)
all.equal(names(out_cyclone$phases), colnames(cpm_quantNormed_fucci))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>all.equal(names(out_cyclone$phases), rownames(pdata_fucci))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>out_cyclone$cell_state &lt;- pdata_fucci$cell_state[match(names(out_cyclone$phases),
                                                       rownames(pdata_fucci))]
out_cyclone$phases &lt;- factor(out_cyclone$phases,
                             levels = c(&quot;G1&quot;, &quot;S&quot;, &quot;G2M&quot;))
table(pdata_fucci$cell_state, out_cyclone$phases)</code></pre>
<pre><code>     
      G1  S G2M
  G1  91  0   0
  S    0 78   2
  G2M  0  0  76</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scran_1.10.2                Seurat_3.1.0               
 [3] forcats_0.3.0               stringr_1.3.1              
 [5] dplyr_0.8.0.1               purrr_0.3.2                
 [7] readr_1.3.1                 tidyr_0.8.3                
 [9] tibble_2.1.1                ggplot2_3.2.1              
[11] tidyverse_1.2.1             yarrr_0.1.5                
[13] circlize_0.4.8              BayesFactor_0.9.12-4.2     
[15] Matrix_1.2-17               coda_0.19-2                
[17] jpeg_0.1-8                  peco_0.99.10               
[19] SingleCellExperiment_1.4.1  SummarizedExperiment_1.12.0
[21] DelayedArray_0.8.0          BiocParallel_1.16.0        
[23] matrixStats_0.55.0          Biobase_2.42.0             
[25] GenomicRanges_1.34.0        GenomeInfoDb_1.18.1        
[27] IRanges_2.16.0              S4Vectors_0.20.1           
[29] BiocGenerics_0.28.0        

loaded via a namespace (and not attached):
  [1] reticulate_1.10          R.utils_2.7.0           
  [3] tidyselect_0.2.5         htmlwidgets_1.3         
  [5] grid_3.5.1               Rtsne_0.15              
  [7] munsell_0.5.0            codetools_0.2-15        
  [9] ica_1.0-2                statmod_1.4.30          
 [11] future_1.14.0            withr_2.1.2             
 [13] colorspace_1.3-2         knitr_1.20              
 [15] rstudioapi_0.10          ROCR_1.0-7              
 [17] gbRd_0.4-11              listenv_0.7.0           
 [19] Rdpack_0.11-0            labeling_0.3            
 [21] git2r_0.26.1             GenomeInfoDbData_1.2.0  
 [23] rhdf5_2.26.2             rprojroot_1.3-2         
 [25] generics_0.0.2           circular_0.4-93         
 [27] R6_2.4.0                 doParallel_1.0.14       
 [29] ggbeeswarm_0.6.0         rsvd_1.0.0              
 [31] conicfit_1.0.4           locfit_1.5-9.1          
 [33] bitops_1.0-6             assertthat_0.2.1        
 [35] promises_1.0.1           SDMTools_1.1-221.1      
 [37] scales_1.0.0             beeswarm_0.2.3          
 [39] gtable_0.2.0             npsurv_0.4-0            
 [41] globals_0.12.4           workflowr_1.6.0         
 [43] rlang_0.4.0              MatrixModels_0.4-1      
 [45] GlobalOptions_0.1.0      splines_3.5.1           
 [47] lazyeval_0.2.1           genlasso_1.4            
 [49] broom_0.5.1              yaml_2.2.0              
 [51] reshape2_1.4.3           modelr_0.1.2            
 [53] backports_1.1.2          httpuv_1.4.5            
 [55] tools_3.5.1              gplots_3.0.1            
 [57] RColorBrewer_1.1-2       dynamicTreeCut_1.63-1   
 [59] ggridges_0.5.1           Rcpp_1.0.3              
 [61] plyr_1.8.4               zlibbioc_1.28.0         
 [63] RCurl_1.95-4.11          pbapply_1.3-4           
 [65] viridis_0.5.1            cowplot_0.9.4           
 [67] zoo_1.8-4                haven_1.1.2             
 [69] ggrepel_0.8.0            cluster_2.0.7-1         
 [71] fs_1.3.1                 magrittr_1.5            
 [73] data.table_1.12.0        lmtest_0.9-36           
 [75] RANN_2.6.1               mvtnorm_1.0-11          
 [77] whisker_0.3-2            fitdistrplus_1.0-14     
 [79] hms_0.4.2                lsei_1.2-0              
 [81] evaluate_0.12            readxl_1.1.0            
 [83] gridExtra_2.3            shape_1.4.4             
 [85] compiler_3.5.1           scater_1.10.1           
 [87] KernSmooth_2.23-15       crayon_1.3.4            
 [89] R.oo_1.22.0              htmltools_0.3.6         
 [91] later_0.7.5              RcppParallel_4.4.3      
 [93] lubridate_1.7.4          MASS_7.3-51.1           
 [95] boot_1.3-20              wesanderson_0.3.6       
 [97] cli_1.1.0                R.methodsS3_1.7.1       
 [99] gdata_2.18.0             metap_1.1               
[101] igraph_1.2.2             pkgconfig_2.0.3         
[103] plotly_4.8.0             xml2_1.2.0              
[105] foreach_1.4.4            vipor_0.4.5             
[107] XVector_0.22.0           bibtex_0.4.2            
[109] rvest_0.3.2              digest_0.6.20           
[111] sctransform_0.2.0        RcppAnnoy_0.0.11        
[113] pracma_2.2.9             tsne_0.1-3              
[115] rmarkdown_1.10           cellranger_1.1.0        
[117] leiden_0.3.1             edgeR_3.24.0            
[119] uwot_0.1.3               geigen_2.3              
[121] DelayedMatrixStats_1.4.0 gtools_3.8.1            
[123] nlme_3.1-137             jsonlite_1.6            
[125] Rhdf5lib_1.4.3           BiocNeighbors_1.0.0     
[127] limma_3.38.3             viridisLite_0.3.0       
[129] pillar_1.3.1             lattice_0.20-38         
[131] httr_1.3.1               survival_2.43-1         
[133] glue_1.3.0               png_0.1-7               
[135] iterators_1.0.12         stringi_1.2.4           
[137] HDF5Array_1.10.1         caTools_1.17.1.1        
[139] irlba_2.3.3              future.apply_1.3.0      
[141] ape_5.2                 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scran_1.10.2                Seurat_3.1.0               
 [3] forcats_0.3.0               stringr_1.3.1              
 [5] dplyr_0.8.0.1               purrr_0.3.2                
 [7] readr_1.3.1                 tidyr_0.8.3                
 [9] tibble_2.1.1                ggplot2_3.2.1              
[11] tidyverse_1.2.1             yarrr_0.1.5                
[13] circlize_0.4.8              BayesFactor_0.9.12-4.2     
[15] Matrix_1.2-17               coda_0.19-2                
[17] jpeg_0.1-8                  peco_0.99.10               
[19] SingleCellExperiment_1.4.1  SummarizedExperiment_1.12.0
[21] DelayedArray_0.8.0          BiocParallel_1.16.0        
[23] matrixStats_0.55.0          Biobase_2.42.0             
[25] GenomicRanges_1.34.0        GenomeInfoDb_1.18.1        
[27] IRanges_2.16.0              S4Vectors_0.20.1           
[29] BiocGenerics_0.28.0        

loaded via a namespace (and not attached):
  [1] reticulate_1.10          R.utils_2.7.0           
  [3] tidyselect_0.2.5         htmlwidgets_1.3         
  [5] grid_3.5.1               Rtsne_0.15              
  [7] munsell_0.5.0            codetools_0.2-15        
  [9] ica_1.0-2                statmod_1.4.30          
 [11] future_1.14.0            withr_2.1.2             
 [13] colorspace_1.3-2         knitr_1.20              
 [15] rstudioapi_0.10          ROCR_1.0-7              
 [17] gbRd_0.4-11              listenv_0.7.0           
 [19] Rdpack_0.11-0            labeling_0.3            
 [21] git2r_0.26.1             GenomeInfoDbData_1.2.0  
 [23] rhdf5_2.26.2             rprojroot_1.3-2         
 [25] generics_0.0.2           circular_0.4-93         
 [27] R6_2.4.0                 doParallel_1.0.14       
 [29] ggbeeswarm_0.6.0         rsvd_1.0.0              
 [31] conicfit_1.0.4           locfit_1.5-9.1          
 [33] bitops_1.0-6             assertthat_0.2.1        
 [35] promises_1.0.1           SDMTools_1.1-221.1      
 [37] scales_1.0.0             beeswarm_0.2.3          
 [39] gtable_0.2.0             npsurv_0.4-0            
 [41] globals_0.12.4           workflowr_1.6.0         
 [43] rlang_0.4.0              MatrixModels_0.4-1      
 [45] GlobalOptions_0.1.0      splines_3.5.1           
 [47] lazyeval_0.2.1           genlasso_1.4            
 [49] broom_0.5.1              yaml_2.2.0              
 [51] reshape2_1.4.3           modelr_0.1.2            
 [53] backports_1.1.2          httpuv_1.4.5            
 [55] tools_3.5.1              gplots_3.0.1            
 [57] RColorBrewer_1.1-2       dynamicTreeCut_1.63-1   
 [59] ggridges_0.5.1           Rcpp_1.0.3              
 [61] plyr_1.8.4               zlibbioc_1.28.0         
 [63] RCurl_1.95-4.11          pbapply_1.3-4           
 [65] viridis_0.5.1            cowplot_0.9.4           
 [67] zoo_1.8-4                haven_1.1.2             
 [69] ggrepel_0.8.0            cluster_2.0.7-1         
 [71] fs_1.3.1                 magrittr_1.5            
 [73] data.table_1.12.0        lmtest_0.9-36           
 [75] RANN_2.6.1               mvtnorm_1.0-11          
 [77] whisker_0.3-2            fitdistrplus_1.0-14     
 [79] hms_0.4.2                lsei_1.2-0              
 [81] evaluate_0.12            readxl_1.1.0            
 [83] gridExtra_2.3            shape_1.4.4             
 [85] compiler_3.5.1           scater_1.10.1           
 [87] KernSmooth_2.23-15       crayon_1.3.4            
 [89] R.oo_1.22.0              htmltools_0.3.6         
 [91] later_0.7.5              RcppParallel_4.4.3      
 [93] lubridate_1.7.4          MASS_7.3-51.1           
 [95] boot_1.3-20              wesanderson_0.3.6       
 [97] cli_1.1.0                R.methodsS3_1.7.1       
 [99] gdata_2.18.0             metap_1.1               
[101] igraph_1.2.2             pkgconfig_2.0.3         
[103] plotly_4.8.0             xml2_1.2.0              
[105] foreach_1.4.4            vipor_0.4.5             
[107] XVector_0.22.0           bibtex_0.4.2            
[109] rvest_0.3.2              digest_0.6.20           
[111] sctransform_0.2.0        RcppAnnoy_0.0.11        
[113] pracma_2.2.9             tsne_0.1-3              
[115] rmarkdown_1.10           cellranger_1.1.0        
[117] leiden_0.3.1             edgeR_3.24.0            
[119] uwot_0.1.3               geigen_2.3              
[121] DelayedMatrixStats_1.4.0 gtools_3.8.1            
[123] nlme_3.1-137             jsonlite_1.6            
[125] Rhdf5lib_1.4.3           BiocNeighbors_1.0.0     
[127] limma_3.38.3             viridisLite_0.3.0       
[129] pillar_1.3.1             lattice_0.20-38         
[131] httr_1.3.1               survival_2.43-1         
[133] glue_1.3.0               png_0.1-7               
[135] iterators_1.0.12         stringi_1.2.4           
[137] HDF5Array_1.10.1         caTools_1.17.1.1        
[139] irlba_2.3.3              future.apply_1.3.0      
[141] ape_5.2                 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
