---
title: "Access data"
author: "John Blischak, Joyce Hsiao"
output:
  workflowr::wflow_html:
    toc: TRUE
    toc_float: TRUE
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
```


Here are some useful links for how we processed our data.

## Data filtering and normalization

For imaging data:

  * First, we [processed FUCCI and DAPI images](https://jdblischak.github.io/fucci-seq/images-process.html) and computed background-corrected sum intensity for each cell image.
  * Second, we [corrected for C1 batch effects in FUCCI intensity data](https://jhsiao999.github.io/peco-paper/images_batchcorrect.html).


For scRNA-seq data:

  * First, we assessed potential C1 batch effects in [sequencing depth](https://jhsiao999.github.io/peco-paper/totals.html) and [read to molecule conversion rate](https://jhsiao999.github.io/peco-paper/reads-v-molecules.html).
  
  * Second, we identified high quality samples using [sample quality metrics](https://jhsiao999.github.io/peco-paper/sampleqc.html), such as total number of mapped reads, ERCC percentages, etc.
    
  * Third, [we excluded over-expressed and lowly-expressed genes](https://jhsiao999.github.io/peco-paper/gene_filtering.html). 
  
  * Finally, we used [Principal Component Analysis](https://jhsiao999.github.io/peco-paper/pca_tf.html) to examined potential sources of gene expression variation.


To create the final dataset for analysis:

  * We [combined scRNA-seq with image intensity data](https://github.com/jhsiao999/peco-paper/tree/master/code/output-raw-2-final.R).


## Downloading the data files

You have two main options for downloading the data files. First, you can
manually download the individual files by clicking on the links on this page or
navigating to the files in the
[fucci-seq](https://github.com/jdblischak/fucci-seq) GitHub repository. This is
the recommended strategy if you only need a few data files.

Second, you can install [git-lfs](https://git-lfs.github.com/). To handle large
files, we used Git Large File Storage (LFS). This means that the files that you
download with `git clone` are only plain text files that contain identifiers for
the files saved on GitHub's servers. If you want to download all of the data
files at once, you can do this with after you install git-lfs.

To install git-lfs, follow their instructions to download, install, and setup
(`git lfs install`). Alternatively, if you use conda, you can install git-lfs
with `conda install -c conda-forge git-lfs`. Once installed, you can download
the latest version of the data files with `git lfs pull`.




